frontend/src/app/page.tsx

```diff
@@ -1,441 +1,131 @@
-import { Route, RouteLap, StationTime } from "./models/route.model";
-import RouteSearch from "./routes/RouteSearch";
-import { Week, Weekdays } from "./models/weekdays.model";
-import Station from "./models/station.model";
-import RouteSearchResult from "./routes/RouteSearchResult";
-import dayjs from "dayjs";
-import "dayjs/locale/en";
-
-const HomePage: React.FC = () => {
-  const [routeResults, setRouteResults] = useState<RouteLap[]>([]);
-  const [originalRoutes, setOriginalRoutes] = useState<Route[]>([]);
-  const [stations, setStations] = useState<Station[]>([]);
-  const [selectedDepartureStation, setSelectedDepartureStation] = useState<
-    string | null
-  >(null);
-  const [selectedArrivalStation, setSelectedArrivalStation] = useState<
-    string | null
-  >(null);
-  const [dateOfDeparture, setDateOfDeparture] =
-    React.useState<dayjs.Dayjs | null>(dayjs());
-  const [error, setError] = useState<string | null>(null);
-  const [hasSearched, setHasSearched] = useState<boolean>(false);
-  const [tempDepartureStation, setTempDepartureStation] = useState<
-    string | null
-  >(null);
-  const [tempArrivalStation, setTempArrivalStation] = useState<string | null>(
-    null
-  );
-  const [tempDepartureDate, setTempDepartureDate] = useState<number | null>(
-    null
-  );
-  const [fixedIsToday, setFixedIsToday] = useState<boolean>(false);
-  const [pastDepartures, setPastDepartures] = useState<RouteLap[]>([]);
-
-  const isToday = (date: dayjs.Dayjs | null): boolean => {
-    return date ? date.isSame(dayjs(), "day") : false;
-  };
-
-  const [isPastDeparturesExpanded, setIsPastDeparturesExpanded] =
-    useState(false);
-
-  const togglePastDepartures = () => {
-    setIsPastDeparturesExpanded((prevState) => !prevState);
-  };
-
-  const handleDateChange = (date: dayjs.Dayjs | null) => {
-    setDateOfDeparture(date);
-
-    if (date) {
-      const dayIndex = (date.day() + 6) % 7;
-
-      setTempDepartureDate(dayIndex);
-    }
-  };
+import { useParams } from "next/navigation";
+import { Card } from "@nextui-org/react";
+import AgencyService from "../../services/agency.service";
+import { Agency } from "../../models/agency.model";
+import { City } from "../../models/city.model";
+import CityService from "../../services/city.service";
+import { Button } from "@shadcn/ui";
+
+const AgencyPage: React.FC = () => {
+  const [agency, setAgency] = useState<Agency | null>(null);
+  const [city, setCity] = useState<City | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  const { id } = useParams();
 
   useEffect(() => {
-    const today = (new Date().getDay() + 6) % 7;
-    setTempDepartureDate(today);
-
-    const fetchRoutes = async () => {
+    const fetchAgency = async () => {
       try {
-        const fetchedRoutes = await RouteService.getRoutes();
-        setOriginalRoutes(fetchedRoutes);
+        const fetchedAgency = await AgencyService.getAgency(id as string);
```

Comment:
you seem to be adding this agency/city functionality as part of your PRs that should just change the buttons on the homepage. please make sure you understand the requirements before opening any PRs